lib/locker/advisory.rb

```diff
@@ -32,8 +34,13 @@ def self.run(key, options={}, &block)
     def run(&block)
       connection = ActiveRecord::Base.connection_pool.checkout
       connection.transaction :requires_new => true do
+        if @blocking && @block_timeout
+          break_at = Time.now + @block_timeout
+        end
+
         while !get(connection) && @blocking
-          sleep 0.005
+          sleep @block_spin_wait
+          break if break_at && break_at < Time.now
```

Comment:
I would reverse the order on these.  Currently it says, "if we didn't get it, sleep a bit, and if we're now past the timeout, bail" (meaning that last sleep was completely wasted).  If we put the break before the sleep, then it'll always at least check the connection after the sleep.